File: prog/1stPartyLibs/quirrel/quirrel/doc/source/performance/tests/lua/exp.lua
local function expLoop(n)

    local sum = 0
    for i = 0, n do
      sum = sum + math.exp(1./(1.+i))
    end
    return sum
end
loadfile("profile.lua")()

io.write(string.format("\"exp loop\", %.8f, 20\n", profile_it(20, function () expLoop(1000000) end)))

File: prog/1stPartyLibs/quirrel/quirrel/bench/js/exp.js
// exponent test

function expLoop(n) {
	var sum = 0;
	for ( var i=0; i!=n; ++i ) {
		sum += Math.exp(1.0/(1.0+i));
	}
	return sum;
}

// infrastructure

function timeStamp() {
   return Date.now();
}

function profile(tname,cnt,testFn) {
	var t = 100500;
	var count = cnt
	while ( count>0 ) {
		var t0 = timeStamp();
		testFn();
		var t1 = timeStamp();
		var dt = t1 - t0;
		t = Math.min(dt, t);
		count --;
	}
	t /= 1000.0;
	var msg = '"'+tname+'", '+t+', '+cnt;
    print(msg)
}

function performance_tests() {
	profile("exp loop",20,function(){
		expLoop(1000000);
	});
	timeStamp();
}

performance_tests();
